### PR TITLE
create url with sig

### PR DESCRIPTION
**Pull request by vg-arimura**
## Create URL with signature
- This function will be used in sop's redirect link.

### CODE REVIEW

File: lib/WebService/SOP/Auth/V1_1/Util.pm [hunk 1]
```diff
@@ -34,16 +34,46 @@ sub is_signature_valid {
...
-    return if $req_time < ($time - $SIG_VALID_FOR_SEC)
-           or $req_time > ($time + $SIG_VALID_FOR_SEC);
+    return
+        if $req_time < ($time - $SIG_VALID_FOR_SEC)
+        or $req_time > ($time + $SIG_VALID_FOR_SEC);
 
     $sig eq create_signature($params, $app_secret);
 }
 
+sub create_url_with_sig {
```
Comment (vg-arimura) [inline lib/WebService/SOP/Auth/V1_1/Util.pm]: Add this subroutine

File: t/01_util.t [hunk 2]
```diff
@@ -6,134 +6,153 @@ use Test::Pretty;
...
-                '1234',
-            );
-        } 'Fails when string is not JSON';
+            WebService::SOP::Auth::V1_1::Util::is_signature_valid('signature', '{"mal":"formed"',
+                'hogehoge', '1234',);
+        }
+        'Fails when string is not JSON';
+    };
+
+    subtest 'create_url_with_sig' => sub {
```
Comment (vg-arimura) [inline t/01_util.t]: Added this test case.

**Comment by yowcowvg** [comment]
This operation should be done by consumer of the library, not by this library.

An app consuming this library will just need to do:
```perl
use WebService::SOP::Auth::V1_1::Util qw(create_signature);

my %query = ($uri->query_form, time => time);
$query{sig} = create_signature(\%query, 'some-secret-key');
$uri->query_form(\%query);

say $uri->as_string;
```

**Comment by vg-arimura** [comment]
@yowcowvg Ahh, got it.

**Pull request closed without merging**